stdlib/tools/swift-reflection-test/swift-reflection-test.c

```diff
@@ -908,5 +966,9 @@ int main(int argc, char *argv[]) {
   uint16_t Version = swift_reflection_getSupportedMetadataVersion();
   printf("Metadata version: %u\n", Version);
 
-  return doDumpHeapInstance(BinaryFilename);
+  PipeMemoryReader Pipe = createPipeMemoryReader();
+  int ret = doDumpHeapInstance(BinaryFilename, &Pipe);
+  destroyPipeMemoryReader(&Pipe);
```

Comment:
Memory management:  Allocate the `PipeMemoryReader` here on the stack and pass it into `doDumpHeapInstance`.   This was simpler than changing `doDumpHeapInstance` to clean up its own `PipeMemoryReader` (because `doDumpHeapInstance` has a bunch of `return` statements).